services/soap/src/main/java/org/mule/services/soap/client/SoapCxfClient.java

```diff
@@ -247,4 +249,13 @@ private AttachmentRequestEnricher getRequestEnricher(boolean isMtom) {
   private AttachmentResponseEnricher getResponseEnricher(boolean isMtom) {
     return isMtom ? new MtomResponseEnricher(introspecter, loader) : new SoapAttachmentResponseEnricher(introspecter, loader);
   }
+
+  private String parseExceptionDetail(Element detail) {
+    try {
+      return XmlTransformationUtils.nodeToString(detail);
+    } catch (XmlTransformationException e) {
+      // Log this.
+      return null;
```

Comment:
why would you return null instead of failing? If Null is an possibility, shouldn't this return Optional?